### PR TITLE
Crew Manifest Improvements

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
Improves the crew manifest as seen in this screenshot (outdated):
image

## Why It's Good For The Game
See it for yourself.

## Changelog
:cl:
tweak: Improved the crew manifest.
/:cl:

### CODE REVIEW

**Comment by Chaznoodles** [comment]
image

Looks odd because it doesn't follow the table layout in the line above.

**Comment by BraveMole** [comment]
I think rank/name/activity would be better.

And maybe you could sort the ranks by alphabetic order?

**Comment by XSlayer300** [comment]
> image
> 
> Looks odd because it doesn't follow the table layout in the line above. Same for all of the subheadings.
> 
> 'Enlisted Engineering Staff' is a mouthful and can easily just be shortened to Engineering, and doesn't make sense when the CSE is an officer.
> 'Dedicated Medical Support' is a mouthful and can easily just be shortened to Medical.
> 'Command and Control Unit' is a mouthful and can easily just be Command.

I was trying to add fancy names compared to plain "engineering", "medical" or "command", but I think it's just me.


> 'Marine Personnel' doesn't make sense when everyone except the Synth and CL are marines.

Last one is simply for marines who are in a squad, synthethic and CL are in the "aux. support staff" and "misc." respectfully.

File: code/datums/datacore.dm [hunk 1]
```diff
@@ -72,25 +76,30 @@ GLOBAL_DATUM_INIT(datacore, /datum/datacore, new)
...
+	if(length(support) > 0)
+		dat += "<tr><th colspan=3>Auxiliary Support Staff</th></tr>"
+		for(var/name in support)
+			dat += "<tr[even ? " class='alt'" : ""]><td>[name]</td><td>[support[name]]</td><td>[isactive[name]]</td></tr>"
+			even = !even
 	if(length(mar) > 0)
-		dat += "<tr><th colspan=3>Marines</th></tr>"
+		dat += "<tr><th colspan=3>Marine Personnel</th></tr>"
 		for(var/j in list("Alpha","Bravo","Charlie", "Delta"))
 			dat += "<tr><th colspan=3>[j]</th></tr>"
```
Comment (spookydonut) [inline code/datums/datacore.dm:91]: hide empty squads

**Review by spookydonut — CHANGES_REQUESTED** [review]

File: code/datums/datacore.dm (hunk 1 above)
Comment (XSlayer300) [reply code/datums/datacore.dm:91]: That... is something I cannot really do, sadly.

File: code/datums/datacore.dm [hunk 2]
```diff
@@ -72,33 +76,38 @@ GLOBAL_DATUM_INIT(datacore, /datum/datacore, new)
...
+	if(length(support) > 0)
+		dat += "<tr><th colspan=3>Auxiliary Support Staff</th></tr>"
+		for(var/name in support)
+			dat += "<tr[even ? " class='alt'" : ""]><td>[support[name]]</td><td>[name]</td><td>[isactive[name]]</td></tr>"
 			even = !even
 	if(length(mar) > 0)
-		dat += "<tr><th colspan=3>Marines</th></tr>"
+		dat += "<tr><th colspan=3>Marine Personnel</th></tr>"
 		for(var/j in list("Alpha","Bravo","Charlie", "Delta"))
 			dat += "<tr><th colspan=3>[j]</th></tr>"
```
Comment (SplinterGP) [inline code/datums/datacore.dm:91]: ty this for spooky suggestion(remember to tab the rest below this as well
```suggestion
		for(var/j in list("Alpha","Bravo","Charlie", "Delta"))
			if(length(squads[j]
				dat += "<tr><th colspan=3>[j]</th></tr>"
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**